src/internal/connector/onedrive/restore.go

```diff
@@ -118,50 +146,130 @@ func RestoreCollection(
 		select {
 		case <-ctx.Done():
 			errUpdater("context canceled", ctx.Err())
-			return metrics, true
+			return metrics, folderPerms, true
 
 		case itemData, ok := <-items:
 			if !ok {
-				return metrics, false
-			}
-			metrics.Objects++
-
-			metrics.TotalBytes += int64(len(copyBuffer))
-
-			itemInfo, err := restoreItem(ctx,
-				service,
-				itemData,
-				drivePath.DriveID,
-				restoreFolderID,
-				copyBuffer,
-				source)
-			if err != nil {
-				errUpdater(itemData.UUID(), err)
-				continue
+				return metrics, folderPerms, false
 			}
 
 			itemPath, err := dc.FullPath().Append(itemData.UUID(), true)
 			if err != nil {
 				logger.Ctx(ctx).DPanicw("transforming item to full path", "error", err)
 				errUpdater(itemData.UUID(), err)
-
 				continue
 			}
 
-			deets.Add(
-				itemPath.String(),
-				itemPath.ShortRef(),
-				"",
-				true,
-				itemInfo)
+			if source == OneDriveSource {
+				name := itemData.UUID()
+				if strings.HasSuffix(name, ".data") {
+					metrics.Objects++
+					metrics.TotalBytes += int64(len(copyBuffer))
+					trimmedName := name[:len(name)-5]
```

Comment:
nit: this should use the defined constant (i.e. `len(dataSuffix)`) or a strings package function if one exists (i.e. `TrimSuffix(dataSuffix)`)